src/core/segment_reader.rs

```diff
@@ -330,6 +447,75 @@ impl SegmentReader {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+/// FieldMetadata
+pub struct FieldMetadata {
+    /// The field name
+    pub field_name: String,
+    /// The field type
+    pub typ: Type,
+    /// Is the field indexed for search
+    pub indexed: bool,
+    /// Is the field stored in the doc store
+    pub stored: bool,
+    /// Is the field stored in the columnar storage
+    pub fast: bool,
+}
+impl PartialOrd for FieldMetadata {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for FieldMetadata {
+    fn cmp(&self, other: &Self) -> Ordering {
+        match self.field_name.cmp(&other.field_name) {
+            Ordering::Equal => self.typ.cmp(&other.typ),
```

Comment:
Let's avoid inconsistent Ord / PartialOrd / Eq/ PartialEq / Hash implementation. This is a very common footgun.
(Here your Ord impl is inconsistent with Eq)

Reply:
I'll switch to a derive with a comment to not reorder